providers/src/airflow/providers/amazon/aws/hooks/eks.py

```diff
@@ -612,9 +614,11 @@ def generate_config_file(
     def fetch_access_token_for_cluster(self, eks_cluster_name: str) -> str:
         session = self.get_session()
         service_id = self.conn.meta.service_model.service_id
-        sts_url = (
-            f"https://sts.{session.region_name}.amazonaws.com/?Action=GetCallerIdentity&Version=2011-06-15"
-        )
+        os.environ["AWS_STS_REGIONAL_ENDPOINTS"] = "regional"
```

Comment:
Maybe leave a comment to explain why we need this?

Reply:
Done!